core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -842,6 +848,114 @@ class ControllerApis(val requestChannel: RequestChannel,
       }
   }
 
+  def allowTokenRequests(request: RequestChannel.Request): Boolean = {
+    val protocol = request.context.securityProtocol
+    if (request.context.principal.tokenAuthenticated ||
+      // We have to allow PLAINTEXT for the controller Apis
+      // protocol == SecurityProtocol.PLAINTEXT ||
```

Comment:
Can we add a check if this is forwarded request then only PLAINTEXT allowed. We still need to disallow for direct  PLAINTEXT requets

Reply:
Yes and done!